src/amb_kf.c

```diff
@@ -531,7 +531,7 @@ s32 find_index_of_element_in_u8s(const u32 num_elements, const u8 x, const u8 *l
       return i;
     }
   }
-  return -1;
+  assert(false);
```

Comment:
I feel like this is a pretty obvious way the function can fail and an assert may be a bit heavy handed here. I think its best to let the caller handle it explicitly. I see the point that sometimes they don't and we get bugs but I'm not sure that making it a fatal error is the best solution. I think we should encourage the caller to be more diligent.